src/Build/BackEnd/Components/ProjectCache/ProjectCacheService.cs

```diff
@@ -591,7 +597,10 @@ public async Task ShutDown()
 
                 await _projectCachePlugin.EndBuildAsync(pluginLogger, _cancellationToken);
 
-                if (pluginLogger.HasLoggedErrors)
+                /* Mismatches between the current plugin state and expected state can occur when cancelled. Under normal circumstances we would want those mismatches to be
+                 * logged as an error. However, in this case the mismatch is caused by the cancellation and should be ignored.
+                */
+                if (pluginLogger.HasLoggedErrors && !_cancellationToken.IsCancellationRequested)
```

Comment:
Not sure this is right. If there are errors during shutdown, even because of cancellation, that seems like an error to me.